### DISCUSSION TITLE
C/C++ LSP doesn't grey out or otherwise indicate sections disabled by preprocessor

### DISCUSSION THREAD

**Discussion by Lunaphied** [helix-editor/helix#11577] [General] [answered]
When using the clangd LSP with Neovim I'm able to see that sections of code that are disabled by the preprocess are disabled.

One example that I hit is `#ifdef __LINUX__` on a Mac. Under Neovim it's very clear that this section is not being compiled and thus the LSP not being functional in that section is expected.

However, in Helix this section appears the same as any other code, except the LSP doesn't work and it's not obvious what's going on without manually searching through the file.

Notably this *does* require using the LSP for highlighting, Tree-sitter or similar alone are not enough (though theoretically that could not be made to work)

**Comment by pascalkuthe** [comment, marked as answer]
duplicate of  #5589

altough clangd could just add diagnostics with the disabled tag like rust-analyzer does which would be much clearer and broader supported. 

In general we do not w3ant to support full lsp highlighting helix is first and foremost tree-sitter based editor
